internal/core/services/room_manager/room_manager.go

```diff
@@ -98,9 +98,36 @@ func (m *RoomManager) DeleteRoom(ctx context.Context, gameRoom *game_room.GameRo
 		return fmt.Errorf("failed to delete instance on the runtime: %w", err)
 	}
 
-	err = m.SetRoomStatus(ctx, gameRoom, game_room.GameStatusTerminating)
+	newGameRoom := *gameRoom
+	newGameRoom.Status = game_room.GameStatusTerminating
+
+	err = m.validateRoomStatusTransition(gameRoom.Status, newGameRoom.Status)
+	if err != nil {
+		return fmt.Errorf("failed when validating game room status transition: %w", err)
+	}
+
+	err = m.roomStorage.UpdateRoom(ctx, &newGameRoom)
+	if err != nil {
+		return fmt.Errorf("failed when updating game room in storage: %w", err)
+	}
+
+	return nil
+}
+
+func (m *RoomManager) UpdateRoom(ctx context.Context, gameRoom *game_room.GameRoom) error {
+	currentGameRoom, err := m.roomStorage.GetRoom(ctx, gameRoom.SchedulerID, gameRoom.ID)
+	if err != nil {
+		return fmt.Errorf("unable to fetch game room from storage: %w", err)
+	}
+
+	err = m.validateRoomStatusTransition(currentGameRoom.Status, gameRoom.Status)
+	if err != nil {
+		return fmt.Errorf("failed when validating game room status transition: %w", err)
+	}
+
+	err = m.roomStorage.UpdateRoom(ctx, gameRoom)
```

Comment:
Are we expecting the other values to be already filled? (`LastPingAt` and `Metadata`)

Reply:
In this case, yes. As I understood, those data will be provided in ping request, right?

Reply:
`Metadata`, yes. The `LastPingAt` is when we receive the ping request, not sure where we're going to "generate it", me it makes sense to be here, WDYT?

Reply:
It makes sense to LastPingAt be generated here, definitely!